### PR TITLE
WIKI-1448: fix the problem of the wiki uri in a space

### PR DESCRIPTION
**Pull request by AzmiTouil**
Fix the problem of wiki URI in space, when we remove then readd wiki app in space, it comes back under /WikiPortlet instead of /wiki?
With this fix, the added app have a URI ="wiki".

### CODE REVIEW

File: component/core/src/main/java/org/exoplatform/social/core/space/impl/DefaultSpaceApplicationHandler.java [hunk 1]
```diff
@@ -266,10 +266,19 @@ private void activateApplicationClassic(Space space, String appId, String appNam
     } catch (Exception e) {
       LOG.warn("space navigation not found.", e);
       return;
-    }   
-    
-    SpaceApplication spaceApplication = new SpaceApplication();
-    spaceApplication.setPortletName(appId);
+    }
+
```
Comment (thomasdelhomenie) [inline component/core/src/main/java/org/exoplatform/social/core/space/impl/DefaultSpaceApplicationHandler.java]:
```suggestion
  // use the uri defined in the config (if any) for the added application to be sure to keep the same uri when an application is removed then added back
```

Comment (AzmiTouil) [reply component/core/src/main/java/org/exoplatform/social/core/space/impl/DefaultSpaceApplicationHandler.java]: unit test added

File: component/core/src/test/java/org/exoplatform/social/core/space/spi/SpaceServiceTest.java [hunk 2]
```diff
@@ -2357,7 +2363,43 @@ public void testInstallApplication() throws Exception {
...
+    NavigationContext navContext = SpaceUtils.getGroupNavigationContext(space.getGroupId());
+    NodeContext<NodeContext<?>> homeNodeCtx = SpaceUtils.getHomeNodeWithChildren(navContext, space.getUrl());
+    Iterator<NodeContext<?>> navigationNodes = homeNodeCtx.getNodes().iterator();
+    boolean found = false;
+    while(navigationNodes.hasNext()){
+      NodeContext<?> node = navigationNodes.next();
+      if ("dashboard".equals(node.getName())){
+        found = true;
+      }
+    }
```
Comment (thomasdelhomenie) [inline component/core/src/test/java/org/exoplatform/social/core/space/spi/SpaceServiceTest.java]: You can replace this loop by:
```
    boolean found = homeNodeCtx.getNodes().stream()
            .filter(node -> "dashboard".equals(node.getName()))
            .findAny()
            .isPresent();
```

File: component/core/src/main/java/org/exoplatform/social/core/space/impl/SpaceServiceImpl.java [hunk 3]
```diff
@@ -890,9 +890,9 @@ public void activateApplication(Space space, String appId) throws SpaceException
     }
     SpaceApplicationHandler appHandler = getSpaceApplicationHandler(space);
     // Default is removable, or must be added by configuration or support setting for applications.
+    appHandler.activateApplication(space, appId, appName);
```
Comment (thomasdelhomenie) [inline component/core/src/main/java/org/exoplatform/social/core/space/impl/SpaceServiceImpl.java]: The comment line "// Default is removable..." is not on the right line anymore.

Comment (thomasdelhomenie) [inline component/core/src/main/java/org/exoplatform/social/core/space/impl/SpaceServiceImpl.java]: Why is this change required ? Is the application really activated if the space is not updated before ?

Comment (AzmiTouil) [reply component/core/src/main/java/org/exoplatform/social/core/space/impl/SpaceServiceImpl.java]: Agreed

File: component/core/src/test/java/org/exoplatform/social/core/space/spi/SpaceServiceTest.java (hunk 2 above)
Comment (AzmiTouil) [reply component/core/src/test/java/org/exoplatform/social/core/space/spi/SpaceServiceTest.java]: Okay

File: component/core/src/main/java/org/exoplatform/social/core/space/impl/SpaceServiceImpl.java (hunk 3 above)
Comment (AzmiTouil) [reply component/core/src/main/java/org/exoplatform/social/core/space/impl/SpaceServiceImpl.java]: yes, it is activated, in fact, I made this change for the !SpaceUtils.isInstalledApp(space, appId) for to work properly
